packages/system/src/factory.ts

```diff
@@ -0,0 +1,28 @@
+import { DOMElements } from "./system.utils"
+import { StyledOptionsChakra, HTMLChakraComponents, styled } from "./system"
+import { As, ChakraComponent } from "./system.types"
+
+type ChakraFactory = {
+  <T extends As, P = {}>(
+    component: T,
+    options?: StyledOptionsChakra,
+  ): ChakraComponent<T, P>
+}
+
+function factory() {
+  const cache = new Map<DOMElements, ChakraComponent<DOMElements>>()
+
+  return new Proxy(styled, {
+    apply(target, thisArg, argArray: [DOMElements, StyledOptionsChakra]) {
+      return styled(...argArray)
+    },
```

Comment:
Any reason this is needed?

Reply:
To allow `chakra("div")` or `chakra(AnotherComponent)` besides `chakra.div` :)

Reply:
I see what you mean, but because the Proxy was created from `styled`, my guess is that we may not need this `apply`. Did you try removing the apply and testing it already?